code/modules/shuttle/white_ship.dm

```diff
@@ -3,4 +3,429 @@
+	var/datum/action/innate/engi_ship/camera_off/off = new
+	var/datum/action/innate/engi_ship/switch_mode/switch_mode_action = new //Action for switching the RCD's build modes
+	var/datum/action/innate/engi_ship/build/build_action = new //Action for using the RCD
+	var/datum/action/innate/engi_ship/airlock_type/airlock_mode_action = new //Action for setting the airlock type
+	var/datum/action/innate/engi_ship/forcefield/shield = new // Action for placing forcefields
+	var/datum/action/innate/engi_ship/recaller/rec = new // BRING HIM HOME
+	var/mob/camera/aiEye/construction/eyeobj
+
+/obj/machinery/computer/ship_construction/attack_hand(mob/user)
+	if(!eyeobj)
+		eyeobj = new(get_turf(src))
+		eyeobj.origin = src
+	GrantActions(user)
+	user.remote_control = eyeobj
+	eyeobj.eye_user = user
+	user.reset_perspective(eyeobj)
+	user.sight |= SEE_TURFS
+	user.update_sight()
+
+/obj/machinery/computer/ship_construction/New()
+	..()
+	RCD = new /obj/item/weapon/rcd/internal(src)
+	FF = new /obj/item/device/forcefield/mounted(src)
+
+/obj/machinery/computer/ship_construction/Destroy()
+	qdel(RCD)
+	qdel(FF)
+	return ..()
+
+/obj/machinery/computer/ship_construction/proc/GrantActions(mob/living/user)
+	off.Grant(user, src)
+	switch_mode_action.Grant(user, src)
+	build_action.Grant(user, src)
+	airlock_mode_action.Grant(user, src)
+	shield.Grant(user, src)
+	rec.Grant(user, src)
+
+/obj/machinery/computer/ship_construction/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/rcd_ammo) || istype(W, /obj/item/stack/sheet))
+		RCD.attackby(W, user, params) //If trying to feed the console more materials, pass it along to the RCD.
+	else
+		return ..()
+
+
+
+/mob/camera/aiEye/construction
+	name = "nss dauntless holo-drone"
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "construction_drone"
+	var/sprint = 10
+	var/cooldown = 0
+	var/acceleration = 1
+	invisibility = INVISIBILITY_MAXIMUM
+	var/obj/machinery/computer/ship_construction/origin
+	var/mob/living/eye_user
+
+/mob/camera/aiEye/construction/Destroy()
+	eye_user = null
+	origin = null
+	return ..()
+
+/mob/camera/aiEye/construction/GetViewerClient()
+	if(eye_user)
+		return eye_user.client
+	return null
+
+/mob/camera/aiEye/construction/setLoc(T)
+	if(eye_user)
+		if(!isturf(eye_user.loc))
+			return
+		T = get_turf(T)
+		loc = T
+
+/mob/camera/aiEye/construction/relaymove(mob/user,direct)
+	var/initial = initial(sprint)
+	var/max_sprint = 50
+
+	if(cooldown && cooldown < world.timeofday) // 3 seconds
+		sprint = initial
+
+	for(var/i = 0; i < max(sprint, initial); i += 20)
+		var/turf/step = get_turf(get_step(src, direct))
+		if(step && (get_dist(origin, step) < 75))
+			src.setLoc(step)
+		else
+			playsound(origin, 'sound/machines/buzz-sigh.ogg', 60, 1)
+
+	cooldown = world.timeofday + 5
+	if(acceleration)
+		sprint = min(sprint + 0.5, max_sprint)
+	else
+		sprint = initial
+
+
+
+
+/datum/action/innate/engi_ship //Parent aux base action
+	var/mob/living/C //Mob using the action

```

Comment:
What is it with you and not learning how to use frameworks before you use them?
First riding datums, now action datums!

Spoilers, this var already exists, it's the action's ```owner``` variable.